src/storage/lock_manager/deadlock.rs

```diff
@@ -25,60 +25,114 @@ use tikv_util::time;
 use tokio_core::reactor::Handle;
 use tokio_timer::Interval;
 
-/// `Locks` is a set of locks belonging to one transaction.
-struct Locks {
-    ts: u64,
-    hashes: Vec<u64>,
-    last_detect_time: time::Instant,
+/// Used to detect the deadlock of wait-for-lock in the cluster.
+pub struct DetectTable {
+    /// Keeps the DAG of wait-for-lock. Every edge from `txn_ts` to `waited_txns` has a survival time -- `ttl`.
+    /// When checking the deadlock, if the ttl has elpased, the corresponding edge will be removed.
+    /// `last_detect_ts` is the start time of the edge. `Detect` requests will refresh it.
+    // txn_ts => waited_txn,which means the txn_ts is waiting for the waited_txns
+    wait_for_map: HashMap<u64, WaitForTxns>,
+
+    /// The ttl of every edge.
+    ttl: time::Duration,
+
+    /// The time of last `active_expire`.
+    last_active_expire: time::Instant,
+}
+
+/// `WaitForTxns` maintain a transaction list.
+#[derive(Default)]
+struct WaitForTxns {
```

Comment:
Emmm why not keep them above `DetectTable`, rather than putting them between  `DetectTable` and `impl DetectTabke`?